lib/tools/adb-emu-commands.js

```diff
@@ -5,42 +5,55 @@ import path from 'path';
 
 let emuMethods = {};
 
+emuMethods.getEmulators = async function() {
+  if (_.isArray(this.emulators) &&
+      this.emulators.length) {
+    return this.emulators;
+  }
+  this.emulators = await this.getConnectedEmulators();
+  return this.emulators;
+};
+
 emuMethods.isEmulatorConnected = async function (udid) {
-  let emulators = await this.getConnectedEmulators();
+  let emulators = await this.getEmulators();
   if (!emulators.length) {
     return false;
   }
-  for (let emulator of emulators) {
-    if (emulator.udid === udid) {
-      return true;
+  return !!_.find(emulators, (x) => x && x.udid === udid);
+};
+
+emuMethods.checkEmulatorConnected = async function (udid) {
```

Comment:
still a bit confused about this method. what do we want to do if `udid` is not defined? just check that _some_ emulator(s) are connected? how does that help?